### PR TITLE
{numlib}[ictce/5.4.0, intel/2015a] PETSc 3.5.3 and 3.6.1(Complex support)

### PR DESCRIPTION
**Pull request by jordiblasco**
These new easyconfigs relies on the commit in the easyblock
https://github.com/hpcugent/easybuild-easyblocks/pull/661

Please, note that the following packages are not available on the new build because they are not coded for complex support.
- Boost
- FIAT
- ScientificPython
- Hypre

### CODE REVIEW

**Comment by boegel** [comment]
Jenkins: ok to test

File: easybuild/easyconfigs/p/PETSc/PETSc-3.5.3-zlibfix.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# due to easybuild provided zlib. The patch filters out lines contaning "no version information available"
+# from stderr in case of ar. PETSc decides based on the extistence of sderr whether a given program failed or not.
+# B. Hajgato August 7th 2014
+--- config/BuildSystem/config/setCompilers.py.orig	2014-06-30 20:42:41.000000000 +0200
++++ config/BuildSystem/config/setCompilers.py	2014-08-07 15:20:31.999008215 +0200
+@@ -1115,6 +1115,7 @@
+     arcUnix    = os.path.join(self.tmpDir, 'libconf1.a')
+     arcWindows = os.path.join(self.tmpDir, 'libconf1.lib')
+     def checkArchive(command, status, output, error):
++      error = '\n'.join([errx for errx in error.split('\n') if "no version information available" not in errx])
```
Comment (boegel) [inline easybuild/easyconfigs/p/PETSc/PETSc-3.5.3-zlibfix.patch]: this patch is probably no longer needed, the zlib easyconfigs have been fixed

right @wpoely86?

File: easybuild/easyconfigs/p/PETSc/PETSc-3.6.1-zlibfix.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+--- config/BuildSystem/config/setCompilers.py.orig	2015-07-23 03:22:46.000000000 +1200
++++ config/BuildSystem/config/setCompilers.py	2015-08-13 11:26:45.043731899 +1200
+@@ -1138,6 +1138,7 @@
+     arcUnix    = os.path.join(self.tmpDir, 'libconf1.a')
+     arcWindows = os.path.join(self.tmpDir, 'libconf1.lib')
+     def checkArchive(command, status, output, error):
++      error = '\n'.join([errx for errx in error.split('\n') if "no version information available" not in errx])
```
Comment (boegel) [inline easybuild/easyconfigs/p/PETSc/PETSc-3.6.1-zlibfix.patch]: patch no longer needed?

File: easybuild/easyconfigs/p/PETSc/PETSc-3.5.3-zlibfix.patch (hunk 1 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/PETSc/PETSc-3.5.3-zlibfix.patch]: yes, you can drop it. But a recompile of zlib is required to fix it...

File: easybuild/easyconfigs/p/PETSc/PETSc-3.6.1-zlibfix.patch (hunk 2 above)
Comment (wpoely86) [inline easybuild/easyconfigs/p/PETSc/PETSc-3.6.1-zlibfix.patch]: yes

**Comment by boegel** [comment]
@jordiblasco: lots of missing easyconfigs:

```
'Irresolvable dependencies encountered: METIS/5.1.0-ictce-5.4.0, SCOTCH/6.0.0_esmumps-intel-2015a, ParMETIS/4.0.3-ictce-5.4.0, SCOTCH/6.0.0_esmumps-ictce-5.4.0, SuiteSparse/4.2.1-ictce-5.4.0-ParMETIS-4.0.3, CMake/3.0.0-ictce-5.4.0'
```

**Comment by migueldiascosta** [comment]
@jordiblasco, this PR is being closed for the following reason(s): no activity for > 6 months, obsoleted by more recent PRs.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**